scipy/sparse/linalg/_dsolve/_superluobject.c

```diff
@@ -105,10 +106,159 @@ static PyObject *SuperLU_solve(SuperLUObject * self, PyObject * args,
     return NULL;
 }
 
+
+static bool is_empty_matrix(const SuperMatrix * A)
+{
+    return (A->nrow == 0 && A->ncol == 0);
+}
+
+
+static PyObject *SuperLU_cond1est(SuperLUObject * self, PyObject * args,
+                               PyObject * kwds)
+{
+    static char *kwlist[] = { "ord", NULL };
+    PyObject* volatile ord_obj = NULL;  /* hold the "ord" argument */
+
+    volatile jmp_buf *jmpbuf_ptr;
+    SLU_BEGIN_THREADS_DEF;
+
+    if (!CHECK_SLU_TYPE(self->type)) {
+        PyErr_SetString(PyExc_ValueError, "unsupported data type");
+        return NULL;
+    }
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "|O", kwlist, &ord_obj))
+        return NULL;
+
+    /* Parse the "ord" argument, which specifies the norm to use.
+     * If not given, we default to '1' (the 1-norm).
+     */
+    char norm_c;
+
+    if (ord_obj == NULL || ord_obj == Py_None) {
+        norm_c = '1';
+    } else if (PyLong_Check(ord_obj)) {  /* check if it's an integer */
+        long ord = PyLong_AsLong(ord_obj);
+        if (ord == -1 && PyErr_Occurred()) {
+            return NULL;  /* conversion error */
+        }
+        if (ord == 1) {
+            norm_c = '1';
+        } else {
+            PyErr_SetString(PyExc_ValueError, "ord must be 1 or np.inf");
+            return NULL;
+        }
+    } else if (PyFloat_Check(ord_obj)) {  /* check if it's a float (np.inf) */
+        double ord = PyFloat_AsDouble(ord_obj);
+        if (ord == -1.0 && PyErr_Occurred()) {
+            return NULL;  /* conversion error */
+        }
+        if (ord == NPY_INFINITY) {
```

Comment:
I think norm1est is OK with getting only the 1-norm so you don't need to perform this norm detection dance. Infinity norm can be obtained by the transpose of the sparse array anyways.